src/charm.py

```diff
@@ -150,6 +150,15 @@ def _on_pgbouncer_pebble_ready(self, event: PebbleReadyEvent) -> None:
             event.defer()
             return
 
+        tls_enabled = all(self.tls.get_tls_files())
+        if self.model.relations.get("certificates", []) and not tls_enabled:
+            self.unit.status = WaitingStatus("Waiting for certificates")
+            event.defer()
+            return
```

Comment:
If a new unit is started with an existing `certificates` relation, defer until the certs become available.

Reply:
Can you please add 
> log.debug("Defering as certificates is not yet pupulated for existing certificates relation")

The status "Waiting for certificates" is great, but when it stuck for unknown reason, it is good to see by users the reason. Also this GH comment will be unnecessary then. :-p